src/platform/input/game-pads.js

```diff
@@ -134,111 +935,142 @@ class GamePads {
+        return this.current[orderIndex]?.wasPressed(button) || false;
     }
 
     /**
      * Returns true if the button was released since the last frame.
      *
-     * @param {number} index - The index of the pad to check, use constants {@link PAD_1},
-     * {@link PAD_2}, etc.
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
      * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
      * @returns {boolean} True if the button was released since the last frame.
      */
-    wasReleased(index, button) {
-        if (!this.current[index]) {
-            return false;
-        }
+    wasReleased(orderIndex, button) {
+        return this.current[orderIndex]?.wasReleased(button) || false;
+    }
 
-        const key = this.current[index].map.buttons[button];
-        const i = pc[key];
+    /**
+     * Get the value of one of the analog axes of the pad.
+     *
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
+     * @param {number} axis - The axis to get the value of, use constants {@link PAD_L_STICK_X}, etc.
+     * @returns {number} The value of the axis between -1 and 1.
+     */
+    getAxis(orderIndex, axis) {
+        return this.current[orderIndex]?.getAxis(axis) || 0;
+    }
 
-        // Previous pad buttons may not have been populated yet
-        // If this is the first time frame a pad has been detected
-        return !this.current[index].pad.buttons[i].pressed && (this.previous[index] && this.previous[index][i]);
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
+     */
+    pulse(orderIndex, intensity, duration, options) {
+        const pad = this.current[orderIndex];
+        return pad ? pad.pulse(intensity, duration, options) : Promise.resolve(false);
     }
 
     /**
-     * Get the value of one of the analogue axes of the pad.
+     * Make all gamepads vibrate.
```

Comment:
As far as I'm aware, vibration is not widely supported in browsers (as of today). Worth putting in a note about what happens when the underlying browser doesn't support the operation?